src/io/inputs.jl

```diff
@@ -166,48 +160,47 @@ function _site_labels(nc_file::NetCDF.NcFile)::Vector{String}
 end
 
 """
-    load_covers(data_fn::String, attr::String, site_data::DataFrame)::NamedDimsArray
+    load_cover(data_fn::String, attr::String, site_data::DataFrame)::YAXArray
+    load_cover(n_species::Int64, n_sites::Int64)::YAXArray
 
 Load initial coral cover data from netCDF.
 """
-function load_covers(data_fn::String, attr::String, site_data::DataFrame)::NamedDimsArray
-    data::NamedDimsArray = load_nc_data(data_fn, attr, site_data)
-    data = NamedDims.rename(data, :covers => :species, :reef_siteid => :sites)
-
-    # Reorder sites to match site_data
-    data = data[sites=Key(site_data[:, :reef_siteid])]
-    data = _convert_abs_to_k(data, site_data)
+function load_cover(data_fn::String, attr::String, site_data::DataFrame)::YAXArray
+    _dim_names_replace = [:covers => :species, :reef_siteid => :sites]
+    data = load_nc_data(data_fn, attr, site_data; dim_names_replace=_dim_names_replace)
 
-    return data
+    return _convert_abs_to_k(data, site_data)
+end
+function load_cover(n_species::Int64, n_sites::Int64)::YAXArray
+    @warn "Using random initial coral cover"
+    axlist = (Dim{:species}(1:(n_species)), Dim{:sites}(1:n_sites))
+    return YAXArray(axlist, rand(Float32, n_species, n_sites))
```

Comment:
We spoke about having a convenience constructor already, just noting it down here.

Reply:
@ConnectedSystems 

I'm still thinking if it's necessary/useful to have this constructor. Let me give an example creating the same array using both `NamedDimsArrays` and `YAXArrays`:

```
nd = NamedDimsArray(rand(Int8, 2,10), row=[:a, :b], col=10:10:100)
ya = YAXArray((Dim{:row}([:a, :b]), Dim{:col}(10:10:100)), rand(Int8, 2, 10))
```

That's for every axes I need to write 7 characters more (`Dim{:}()` versus `=`) with YAXArray syntax. So, if we are creating an array with two axis maybe with `NamedDimsArrays` we wouldn't need more than one line and with `YAXArrays` we would need two (or more, depending on the formatting). But for less or for more than that, I think it's almost the same: if we have 4 dimensions, even with `NamedDimsArrays` it's better (for readability) to create the axlist, the data, and put it all together when creating the array.

ALSO 

1. `YAXArrays` allows us to also pass a metadata param (`props`), and with that even with the constructor it will be better to create these things (data, axlist and props) separately and use it all to create the array. 
2. We will need to give support to this constructor, test it, etc, so, unless it really helps, I tend to think we don't need it.


In any case, I would like to hear your thoughts on that before I do anything :)

Reply:
My thinking regarding suggesting an alternate constructor is about usability and convenience.

I don't suggest having a constructor for every single use case, just the common ones. If there are more specialised contexts where it makes sense to create YAXArrays directly, then we should by all means do so.

If there is a situation where you need to create a set of YAXArrays with named rows and columns as illustrated by the example above, then typing `Dim{:row}([:a, :b])` over and over again seems really annoying, repetitive, and redundant. I'd much prefer the below **for the specific use case you have**.

```julia
function RowColYAX(data, rows, cols; kwargs...)
    return YAXArray((Dim{:row}(rows), Dim{:col}(cols), data; kwargs...))
end

# ... in different locations around the codebase ...
ya = RowColYAX(rand(Int8, 2, 10), [:a, :b], 10:10:100)
ya2 = RowColYAX(rand(Int8, 2, 10), [:c, :d], 10:100:1000)
ya3 = RowColYAX(rand(Int8, 3, 10), [:x, :y, :z], 1:10)

# I'd prefer having to use the above than the below

ya = YAXArray((Dim{:row}([:a, :b]), Dim{:col}(10:10:100)), rand(Int8, 2, 10))
ya2 = YAXArray((Dim{:row}([:c, :d]), Dim{:col}(10:100:1000)), rand(Int8, 2, 10))
ya3 = YAXArray((Dim{:row}([:x, :y, :z]), Dim{:col}(10:10:100)), rand(Int8, 3, 10))
```

Now if you're saying there are no common use cases, then yes I agree, there's no point in having another generic constructor.

Reply:
I have created the DataCube and ZeroDataCube constructors for some common cases. Let me know if you have any suggestions (other than allowing passing a `props`, in the future) or if I've addressed your suggestion here. (: